### PR TITLE
Migrate exchangeCodeForToken to node-fetch

### PR DESCRIPTION
**Pull request by TrySound**
Ref https://github.com/nylas/nylas-nodejs/issues/174

Migrating whole library from "request" at once would be quite massive.
In this diff I migrated only exchangeCodeForToken to figure out how to
test and install dependencies.

# License
I confirm that this contribution is made under the terms of the MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by TrySound** [comment]
cc @sdwvit

File: src/nylas.ts [hunk 1]
```diff
@@ -130,38 +132,36 @@ class Nylas {
...
-      return request(options, (error, response, body) => {
-        if (!body || !body['access_token']) {
-          let errorMessage = 'No access token in response';
-          if (body && body.message) errorMessage = body.message;
-          error = error ? error : new Error(errorMessage);
+    const url = new URL(`${this.apiServer}/oauth/token`);
+    url.searchParams.set('client_id', this.clientId);
+    url.searchParams.set('client_secret', this.clientSecret);
+    url.searchParams.set('grant_type', 'authorization_code');
+    url.searchParams.set('code', code);
```
Comment (sdwvit) [inline src/nylas.ts:139]: using URL is an interesting choice here ;) is it just to urlencode querystring? looks good to me, just wanted to point out

File: src/nylas.ts [hunk 2]
```diff
@@ -130,38 +132,36 @@ class Nylas {
...
-        } else {
-          resolve(body['access_token']);
+          return body['access_token'];
+        },
+        error => {
           if (callback) {
-            return callback(null, body['access_token']);
+            callback(error);
           }
+          throw error;
```
Comment (sdwvit) [inline src/nylas.ts:162]: wouldn't it make sense to recreate an Error so it includes current frame in a stacktrace?

File: src/nylas.ts (hunk 1 above)
Comment (TrySound) [reply src/nylas.ts:139]: Browser support makes it more preferred choice over `querystring`

File: src/nylas.ts (hunk 2 above)
Comment (TrySound) [reply src/nylas.ts]: Done

**Comment by TrySound** [comment]
Hi @sdwvit @Gerdie, is it good to go?

**Pull request merged**